src/firefly/js/visualize/draw/CanvasWrapper.jsx

```diff
@@ -88,9 +90,40 @@ class CanvasWrapper extends React.Component {
     constructor(props) {
         super(props);
         this.drawer= null;
+        this.lastDrawLayer= null;
+        
     }
 
-    shouldComponentUpdate(np,ns) { return sCompare(this,np,ns); }
+    // shouldComponentUpdate(np,ns) {
+    //     return sCompare(this,np,ns);
+    // }
+
+
+    shouldComponentUpdate(nProps) {
+
+        var {plot,drawLayer,width,height}= nProps;
+        const p= this.props;
+
+        const update= (width!==p.width || height!==p.height || plot!==p.plot);
+
+        if (!update) {
+            this.updateDrawLayer(nProps);
+        }
```

Comment:
the `update` is for shouldComponentUpdate. Since the draw layer has a canvas it actually needs to update but not via react.  It should redraw on the existing canvas. This is an optimization. Before we force react to go through it dom diffing.  It did not update but we sitll had to draw.   This was react does not waste time,